sources/engine/Stride.Engine/Engine/FlexibleProcessing/ProcessorManager.cs

```diff
@@ -0,0 +1,187 @@
+            {
+                // One component may have multiple different IComponent<,>, we must handle each definition independently
+                // To do so we have to iterate over all the interfaces this component may contain
+                var interfaces = componentType.GetInterfaces();
+                var processorsForThisType = new List<int>();
+                foreach (var type in interfaces)
+                {
+                    if (type.IsGenericType == false)
+                        continue;
+
+                    if (typeof(IComponent<,>) != type.GetGenericTypeDefinition())
+                        continue; // This interface is not IComponent<,>
+
+                    var processorType = type.GenericTypeArguments[0];
+                    if (typeToIndex.TryGetValue(processorType, out var procIndex) == false)
+                    {
+                        processors.Add(new(null, processorType, 0));
+                        typeToIndex[processorType] = procIndex = processors.Count - 1;
+                    }
+
+                    processorsForThisType.Add(procIndex);
+                }
+
+                compTypeToProcessorIndices[componentType] = processorsIndex = processorsForThisType.ToArray();
+            }
+
+            foreach (var procIndex in processorsIndex)
+            {
+                var pData = processors[procIndex];
+                pData.ComponentCount++;
+
+                if (pData.Instance == null) // if the processor hasn't been created yet or was previously removed
+                {
+                    pData.Instance = (IProcessorBase)Activator.CreateInstance(pData.Type)!;
+                    if (pData.Instance is IUpdateProcessor update)
+                        processorsToUpdate.Add(new(update.Order, pData.Type), new(update, new ProfilingKey(GameProfilingKeys.GameUpdate, GetType().Name)));
+                    if (pData.Instance is IDrawProcessor draw)
+                        processorsToDraw.Add(new(draw.Order, pData.Type), new(draw, new ProfilingKey(GameProfilingKeys.GameDraw, GetType().Name)));
+
+                    processors[procIndex] = pData; // Set beforehand just in case SystemAdded adds a component
+                    pData.Instance!.SystemAdded(registry);
+                    pData = processors[procIndex]; // Refresh local for the same reason
+                }
+                else
+                {
+                    processors[procIndex] = pData;
+                }
+
+                pData.Instance!.OnComponentAdded(component);
+            }
+        }
+
+        public void RemoveComponent(EntityComponent _component)
+        {
+            if (_component is IMarkedComponent component == false)
+                return;
+
+            foreach (var procIndex in compTypeToProcessorIndices[component.GetType()])
+            {
+                var pData = processors[procIndex];
```

Comment:
Would accessing the list as Span allow an inline modification here?
`processors.AsSpan()[procIndex].ComponentCount--`

Reply:
Yep, I thought of using that initially but at the time the logic was set up in a different way which benefited from copying the data, now it fits in like a glove, thanks !